### ISSUE TITLE
Infinite loop with some inputs

### ISSUE THREAD

**Issue by lupine** [profanity-im/profanity#641]
I'm running profanity current git inside tmux (with mouse support on), and connecting to it from two different terminals - gnome-terminal and terminator. Occasionally, for reasons I don't understand, some mouse actions result in characters being registered as keypresses. In irssi, these are registered as written text like "[M#"

In profanity, when this happens it seems to confuse the input loop, resulting in hang and 100% CPU utilisation. I've captured a few backtraces here: https://gist.github.com/lupine/62cf62033cbe2deff195

At a guess, it thinks it's partway through a multibyte character and is aggressively reading for the rest of it. 

I haven't discovered a text input when it gets into this state that rescues it, it's a matter of quit and restart.

**Comment by boothj5** [comment]
I'm failing to recreate this issue, I ran tmux (in terminator)  and tried all mouse events.  Could you clarify what you mean by "with mouse support on"?

Thanks.

**Comment by boothj5** [comment]
I've found the tmux setting you're talking about, and set the following in my `~/.tmux.conf`

```
set -g mode-mouse on
```

However still haven't managed to recreate the problem. I did find something that could possibly be related, and have added a fix in master (https://github.com/boothj5/profanity/commit/a02c89a8e8f6abe8935f6cf95e0b016d1ac40a34) which should no longer try to read too many characters if you are in a non-UTF8 environment.

If you still get the issue, the following might help me recreate:

1) Your `tmux.conf`
2) Output of the `locale` command in the terminal
3) Output of the `/charset` command in profanity (recently added).
